analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionalsWithSameCondition.CSharp9.cs

```diff
@@ -52,11 +52,25 @@ public void Foo()
                 f = cond ? new Apple() : new Orange();
             }
 
+            if (f is null) { }
```

Comment:
could you also add some noncompliant tests with type pattern? we only have compliant ones

```
if (f is Apple a) { ; }
if (f is Apple a) { ; } // Noncompliant
if (f is not Apple) { ; }
if (f is not Apple) { ; } // Noncompliant
```
Also for positional pattern you could add [the example](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#positional-pattern) from the docs - I'm not sure if the compiler is smart enough to raise that (if so, we may not want to duplicate what it does)

Also - could you please add tests with property patterns (probably some logic needs to be updated in the equivalence checker)
```
if (a is { Count: >= 5 }) {}
if (a is { Count: >= 5 }) {} // Noncompliant
```
and more complex ones
```
if (a is { Count: >= 5 }) {}
if (a is not { Count: < 5 }) {} // probably FN
```
and
```
if (segment is { Start: { X: 0, Y: 0 } } or { End: { X: 0, Y: 0 } }) {};
if (segment is { Start: { X: 0, Y: 0 } } or { End: { X: 0, Y: 0 } }) {}; // Noncompliant
```

Reply:
For the docs example, you can't have the same switch arm conditions. It raises error:
>Error CS8510 The pattern is unreachable. It has already been handled by a previous arm of the switch expression or it is impossible to match.

Reply:
I'll skip the last one, it's too difficult to scaffold and we have complicated cases already.